client/src/components/RelatedProducts/Cards/AllCards.jsx

```diff
@@ -1,29 +1,41 @@
 /* eslint-disable max-len */
-import React from 'react';
-import { useProd } from './ProdContext';
-import { useThumbnail } from './thumbnailContext';
+import { React, useState } from 'react';
+import { useProd } from '../contexts/ProdContext';
+import { useThumbnail } from '../contexts/thumbnailContext';
 import CarouselRelated from './CarouselR';
 
 function AllCards() {
-  const relatedProds = useProd().relatedInformation;
+  const { relatedInformation } = useProd();
   const thumbnail = useThumbnail();
-  const ratings = useProd().ratingsMeta;
+  const { ratingsMeta } = useProd();
+  const information = [];
 
-  const zippedArray = [];
-
-  if (thumbnail.data && relatedProds.data && ratings) {
-    relatedProds.data.map((item, i) => {
-      zippedArray.push([item.data, thumbnail.data[i].data.results[0].photos[0].thumbnail_url, ratings.data[i].data]);
+  if (thumbnail?.data && relatedInformation?.data && ratingsMeta) {
```

Comment:
This is probably a great candidate for either a future refactor or some comments explaining what this does. Imagine if you ever need to touch this code again in the future, you might not remember what all of this is doing.